src/commonMain/kotlin/fr/acinq/lightning/payment/OfferManager.kt

```diff
@@ -150,7 +150,16 @@ class OfferManager(val nodeParams: NodeParams, val walletParams: WalletParams, v
             else -> {
                 val amount = request.amount ?: (request.offer.amount!! * request.quantity)
                 val preimage = randomBytes32()
-                val pathId = OfferPaymentMetadata.V1(ByteVector32(decrypted.pathId), amount, preimage, request.payerId, request.quantity, currentTimestampMillis()).toPathId(nodeParams.nodePrivateKey)
+                val truncatedPayerNote = request.payerNote?.let {
+                    val encoded = it.encodeToByteArray()
+                    if (encoded.size <= nodeParams.maxPayerNoteLength) {
+                        it
+                    } else {
+                        val charactersToKeep = encoded.take(nodeParams.maxPayerNoteLength - 3).toByteArray().decodeToString().length - 1
+                        it.take(charactersToKeep) + "…"
+                    }
+                }
```

Comment:
Wouldn't it be simpler to truncate the `String` directly, assuming the worst-case scenario where each character uses 4 bytes?

Reply:
I now truncate to 64 characters